importccl: fix ALTER TABLE ... SET DEFAULT / SET NOT NULL

otan opened the pull request:
Based on #53808, only last commit is relevant.

----

Regression introduced in 921c60a0af19226b57274c070cdb5ca5d14a54d7.

Release justification: low risk, high benefit changes to
existing functionality

Release note: None

otan committed:
importccl: fix sqlbase refactor rebase issue with IMPORT PGDUMP

Looks like 8e5e54f9270d7281184282d5946bd6e2ce6e9485 incorrectly modified
the `idx` back to `i` causing a bug. This was not caught
as TestImportData is skipped (I will fix this separately) -- I
tested this with it unskipped (by hand) beforehand.

Release justification: bug fixes and low-risk updates to new functionality

Release note: None

otan committed:
importccl: partially unskip TestImportData

TestImportData seems to have been skipped for a while. Not sure why, but
it has caused regressions to new tests. I've unskipped the ones that
look like they pass.

Release justification: non-production code changes.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53809)
<!-- Reviewable:end -->

dt commented on pkg/ccl/importccl/read_import_pgdump.go:
```diff
@@ -383,8 +383,10 @@ func readPostgresStmt(
+					// we're not interested in this column.
 					if !ok || def.Name != cmd.Column {
-						return errors.Errorf("unsupported %T definition: %s", def, def)
+						continue
```
would it make sense to have a `found` bool and error out if we never find a def with name == cmd.Column? or just assume it won't happen?

otan replied:
yeah, done!

dt reviewed (approved):
good catch! Thanks!

otan committed:
importccl: fix ALTER TABLE ... SET DEFAULT / SET NOT NULL

Regression introduced in 921c60a0af19226b57274c070cdb5ca5d14a54d7.

Release justification: low risk, high benefit changes to
existing functionality

Release note: None

otan commented:
bors r=dt
